### ISSUE TITLE
LSTM/GRU input_size tensor shape errors

### ISSUE THREAD

**Issue by MCplayerFromPRC** [ropas/pytea#10]
Hi, I have met a problem while detecting LSTM tensor shape errors.
The testing file below is runnable and pytea returns correctly.   

```
import torch
import torch.nn as nn
import torch.nn.functional as F


class Model(nn.Module):

    def __init__(self):
        super().__init__()
        self.cnn = nn.Conv2d(3, 1, 3, 1, 1)
        self.rnn = nn.LSTM(32, 64, 1, batch_first=True)
        self.pool = nn.MaxPool2d(2, 2)
        self.fc = nn.Linear(64, 16)

    def forward(self, x):
        x = self.pool(F.relu(self.cnn(x)))
        x = x.view(-1, 32, 32)
        x, _ = self.rnn(x)
        x = x[:, -1, :].squeeze(1)
        x = F.relu(self.fc(x))
        x = F.softmax(x, dim=-1)
        return x


if __name__ == "__main__":
    net = Model()
    x = torch.randn(2, 3, 64, 64)
    y = net(x)
    target = torch.argmax(torch.randn(2, 16), dim=-1)
    loss = F.cross_entropy(y, target.long())
    loss.backward()
    print(y.size())
```

However, If I change `self.rnn = nn.LSTM(32, 64, 1, batch_first=True)` into `self.rnn = nn.LSTM(64, 64, 1, batch_first=True)`, torch will report a RuntimeError: Expected 64, got 32. 
pytea didn't return any CONSTRAINTS information, as it supposed to.

Then I tried to more LSTM input_size shape errors, all failed. Same situation with GRU.
I think it is a bug, because I can detect Conv2d, Linear error successfully.

**Comment by MerHS** [comment]
PyTea internal torch.nn modules (including RNN) are simplified than the original implementation of the PyTorch modules.
I'll update and correct the behavior of it as possible as I can.

**Comment by MCplayerFromPRC** [comment]
Thank you for answering.

**Comment by MerHS** [comment]
Resolved in c536515a5e5947fac8871784323ba7eddc58956d

**Comment by MCplayerFromPRC** [comment]
[c536515](https://github.com/ropas/pytea/commit/c536515a5e5947fac8871784323ba7eddc58956d) works for me. Thanks!
